scripts/west_commands/runners/spi_burn.py

```diff
@@ -0,0 +1,70 @@
+import os
+import time
+import sys
+import subprocess
+
+from runners.core import ZephyrBinaryRunner, RunnerCaps
+
+TELINK_TOOLCHAIN_BASE = os.environ["TELINK_TOOLCHAIN_BASE"]
+SPI_BURN = f"{TELINK_TOOLCHAIN_BASE}/flash/bin/SPI_burn"
+ICEMAN_DIR = f"{TELINK_TOOLCHAIN_BASE}/ice"
+ICEMAN = "./ICEman"
+
+class SpiBurnBinaryRunner(ZephyrBinaryRunner):
+    '''Runner front-end for SPI_burn.'''
+
+    def __init__(self, cfg, addr, eraseall):
+        super().__init__(cfg)
+        self.bin = cfg.bin_file
+        self.addr = addr
+        if eraseall:
+            self.eraseall = "--erase-all"
+        else:
+            self.eraseall = ''
+
+    @classmethod
+    def name(cls):
+        return 'SPI_burn'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        parser.add_argument('--addr', default='0x0', 
+                            help='start flash address to write')
+        parser.add_argument('--erase-all', default=True,
+                            help='frace flash before burn')
+
+    @classmethod
+    def do_create(cls, cfg, args):
+        return SpiBurnBinaryRunner(cfg, args.addr, args.erase_all)
+
+    def do_run(self, command, **kwargs):
+        
+        origin_dir = os.getcwd()
+        try:
+
+            # Go into ICEman dir
+            os.chdir(ICEMAN_DIR)
+
+            # RUN ice in a background
+            cmd_ice_run = [ICEMAN, '-Z', 'v5', '-I', 'aice_sdp.cfg']
```

Comment:
Why is a chdir needed? Can't this just be `cmd_ice_run = [f'{ICEMAN_DIR}/{ICEMAN}', ...]` or so? If so, please don't chdir. If not, please add a comment explaining why it is necessary.

Reply:
> Can't this just be `cmd_ice_run = [f'{ICEMAN_DIR}/{ICEMAN}', ...]`

Or, even better, just use PATH.

Reply:
That is the most tricky part. The ICEman tool shall be run from its base directory, as it looking for it config in the folder from which it is run. In attempt to run it from other dir it returns an error about missing configuration.

Reply:
> That is the most tricky part.

Gotcha. OK, but that is not in my view a reason to require this extra argument. You can still find the tools using PATH and then change directory to where they are installed before running them as needed.